packages/core/lib/middleware/sampling/service_connector.js

```diff
@@ -53,8 +56,7 @@ var ServiceConnector = {
     });
 
     req.on('error', (err) => {
-      this.logger.getLogger().error(`Failed to connect to X-Ray daemon at ${options.hostname}:${options.port} to get sampling rules.`);
-      callback(err);
+      callback(new Error(`Failed to connect to X-Ray daemon at ${options.hostname}:${options.port} to get sampling rules.`));
     });
```

Comment:
We don't need a return statement here since the `callback` is invoked in its own anonymous function. So in this case the function ends and implicitly returns right after L59. [Source](https://nodejs.org/api/stream.html#stream_event_error).

We don't do retries here because this only runs on request errors, which means that the daemon is not running on the given port to accept requests, and retrying will not change that.

Reply:
okay. That's fair.